libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -44,129 +56,150 @@ interface IRushXCommandLineArguments {
   commandArgs: string[];
 }
 
+class ProcessError extends Error {
+  public exitCode: number = 0;
+  public constructor(message: string, exitCode: number) {
+    super(message);
+    this.exitCode = exitCode;
+  }
+}
+
 export class RushXCommandLine {
-  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
-    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
+  public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+    try {
+      const args: IRushXCommandLineArguments = this._getCommandLineArguments();
+      const rushConfiguration: RushConfiguration | undefined = RushConfiguration.tryLoadFromDefaultLocation({
+        showVerbose: false
+      });
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const suppressHooks: boolean = process.env._RUSH_SUPPRESS_HOOKS === '1';
+      const attemptHooks = !suppressHooks && !args.help;
+      if (attemptHooks) {
+        eventHooksManager?.handle(Event.preRushx, args.isDebug, args.ignoreHooks);
+      }
+      // Node.js can sometimes accidentally terminate with a zero exit code  (e.g. for an uncaught
+      // promise exception), so we start with the assumption that the exit code is 1
+      // and set it to 0 only on success.
+      process.exitCode = 1;
+      RushXCommandLine._launchRushXInternal(launcherVersion, options, rushConfiguration, args);
+      process.exitCode = 0;
+
+      if (attemptHooks) {
+        eventHooksManager?.handle(Event.postRushx, args.isDebug, args.ignoreHooks);
+      }
+    } catch (error) {
+      if (error instanceof ProcessError) {
+        process.exitCode = error.exitCode;
+      } else {
+        process.exitCode = 1;
+      }
+      console.error(colors.red('Error: ' + (error as Error).message));
+    }
```

Comment:
```suggestion
      process.exitCode = 1;
      RushXCommandLine._launchRushXInternal(launcherVersion, options, rushConfiguration, args);

      if (attemptHooks) {
        eventHooksManager?.handle(Event.postRushx, args.isDebug, args.ignoreHooks);
      }

      process.exitCode = 0;
    } catch (error) {
      if (error instanceof ProcessError) {
        process.exitCode = error.exitCode;
      }

      console.error(colors.red('Error: ' + (error as Error).message));
    }
```

Reply:
With this change the exit code will be 0 in the following cases:
*  exception trying to load command line arguments
* exception loading the rush configuration
* exception on the starting hook.    
Is that the desired behavior?

Reply:
@iclanton please look at this

Reply:
@nebosite I think you meant exit "code will be 1" not zero.

Looking at the code, in our existing implementation, if an event hook shell command fails, Rush prints an error message but this [does NOT cause](https://github.com/microsoft/rushstack/blob/d95ad804d7abaf67c707513dd50e3c6672ab546c/libraries/rush-lib/src/logic/EventHooksManager.ts#L54) the Rush command to fail.  In other words, the hooks are treated as nonessential commands (e.g. if we're trying to upload telemetry but there is no internet, it's no big deal).

I think that `rushx` hooks should follow that same policy.  If we want to change the policy, we should do so in a separate PR, and consistently for all hooks.

@nebosite Let's keep your original implementation but add a `//` comment explaining why `process.exitCode = 0;` appears on that line, because it does look strange to a casual observer.

Reply:
OK, I think I understand.   To make this behavior work as described, I have added try catch blocks around the spots where the hooks are called.  Any errors coming out will get printed and otherwise ignored.